autoload/ale/highlight.vim

```diff
@@ -49,6 +58,18 @@ endfunction
 
 " Given a loclist for current items to highlight, remove all highlights
 " except these which have matching loclist item entries.
+if s:nvim_api
+
+function! ale#highlight#RemoveHighlights() abort
+    if get(b:, 'ale_highlight_id', 0)
+        let l:buf = bufnr('%')
+        " NOTE: 0, -1 means from 0 line till the end of buffer
+        call nvim_buf_clear_namespace(l:buf, b:ale_highlight_id, 0, -1)
+    endif
+endfunction
+
+else
+
 function! ale#highlight#RemoveHighlights() abort
```

Comment:
I think you should move the version check inside the function here, and define it once. That will be easier to test, and won't cost too much. It looks like the changes here break earlier versions of NeoVim. You might have to check if `nvim_buf_clear_namespace` exists too. it might be that `nvim_buf_add_highlight` exists in some versions where `nvim_buf_clear_namespace` does not exist.